runtime/near-vm-logic/src/logic.rs

```diff
@@ -278,49 +285,61 @@ impl<'a> VMLogic<'a> {
     /// * If string is longer than `max_log_len` returns `BadUtf8`.
     fn get_utf8_string(&mut self, len: u64, ptr: u64) -> Result<String> {
         let mut buf;
+        let max_len = self.config.max_log_len;
         if len != std::u64::MAX {
-            if len > self.config.max_log_len {
+            if len > max_len {
                 return Err(HostError::BadUTF8.into());
             }
             buf = Self::memory_get(self.memory, ptr, len)?;
         } else {
             buf = vec![];
-            for i in 0..=self.config.max_log_len {
-                Self::try_fit_mem(self.memory, ptr, i)?;
+            for i in 0..=max_len {
+                if i == max_len {
```

Comment:
The reason why I moved it below to allow strings of length `max_len` excluding `\0` symbol for null-terminated setting. Cause strings of this size are allowed for given `len == max_len`.

Reply:
got it!